gurubu-client/src/app/components/room/nickname-form.tsx

```diff
@@ -108,10 +113,16 @@ const NicknameForm = ({ joinMode, roomId }: IProps) => {
     window.location.assign(`/room/${response.roomID}`);
   };
 
-  const handleOptionClick = (typeNumber: string) => {
-    setErrorMessage("");
-    setGroomingType(typeNumber);
-  };
+
+  const connectionButtonText = useMemo(() => {
+    if (loading && roomId) {
+      return "Joining Room...";
+    }
+    if (loading && !roomId) {
+      return "Creating Room...";
+    }
+    return roomId ? "Join Room" : "Create Room";
+  }, [loading, roomId])
```

Comment:
Hey @ethsmaa, I think usage of useMemo here couldn't be necessary. The calculation of this function is not expensive. Maybe we can create a normal function and call it in the return section.
[useMemo](https://react.dev/reference/react/useMemo)

Reply:
thanks! i will fix it ✨